semantiva/component_loader/component_loader.py

```diff
@@ -36,41 +27,79 @@ def register_paths(cls, paths: str | List[str]):
-            module_spec = importlib.util.spec_from_file_location(module_name, path)
+    def get_registered_modules(cls) -> Set[str]:
+        """Get list of registered modules"""
+        return cls._registered_modules
 
-            if module_spec is None or not module_spec.loader:
-                continue
+    @classmethod
+    def get_class(cls, class_name: str):
+        """Lookup in registered paths and modules for the class and
+        return its type. It starts with modules and then looks in paths."""
 
-            module = importlib.util.module_from_spec(module_spec)
-            try:
-                module_spec.loader.exec_module(module)
-            except Exception as e:
-                print(f"Error loading module {module_name}: {e}")
-                continue
+        for module_name in cls._registered_modules:
+            class_type = cls._get_class_from_module(module_name, class_name)
+            if class_type is not None:
+                return class_type
 
-            # Check and return the class type
-            class_type = getattr(module, class_name, None)
+        for path in cls._registered_paths:
+            class_type = cls._get_class_from_file(path, class_name)
             if class_type is not None:
                 return class_type
+
         raise ValueError(
-            f"Class '{class_name}' not found in any of the registered paths."
+            f"Class '{class_name}' not found in any of the registered modules and paths."
         )
 
+    @classmethod
+    def _get_class_from_module(cls, module_name: str, class_name: str):
+        """Lookup in registered modules for the class and
+        return its type. If module is not found, return None."""
+
+        try:
+            module = import_module(module_name)
+            class_type = getattr(module, class_name, None)
+            return class_type
+        except ModuleNotFoundError:
+            return None
+
+    @classmethod
+    def _get_class_from_file(cls, file_path: Path, class_name: str):
+        """Lookup in registered paths for the class and return its type."""
+
+        if not file_path.is_file():  # If path does not exist, skip it
+            return None
+
+        module_name = file_path.stem
+        module_spec = importlib.util.spec_from_file_location(module_name, file_path)
+
+        if module_spec is None or not module_spec.loader:
+            return None
+
+        module = importlib.util.module_from_spec(module_spec)
+        try:
+            module_spec.loader.exec_module(module)
+        except Exception as e:
+            print(f"Error loading module {module_name}: {e}")
+            return None
+
+        # Check and return the class type
+        return getattr(module, class_name, None)
+
 
-# Initialize default paths when the class is loaded
-ComponentLoader.initialize_default_paths()
+# Initialize default modules when the class is loaded
+ComponentLoader.initialize_default_modules()
```

Comment:
This initialization is better suited for __init__.py. Unlikely to cause problems.